ONSdigital/eq-author-app
# Homepage Redesign

JIRA ticket: https://collaborate2.ons.gov.uk/jira/browse/EAR-510

The wave tool bring up 12 alert errors on the new homepage due to the underlined text on the questionnaire titles. After going through this with joe, this is the style we are going with regardless of the 12 alert errors on the wave tool.

### What is the context of this PR?

> Homepage redesign 
- built a new questionnaire table component in the components themed folder (based off the old questionnaire table)
- styled the homepage using the mock up
- new pagination component for the questionnaire table.
> Added the new questionnaire table in storybook
> The pagination component was built using these resources for context
-  https://www.freecodecamp.org/news/build-a-custom-pagination-component-in-react/
- https://www.digitalocean.com/community/tutorials/how-to-build-custom-pagination-with-react
>The design of the homepage follows the mock up on the Jira ticket

### How to review

> Check the design follows the mock design of the homepage on the Jira ticket
> Check the access column sorts correctly ( Sorting by Editor/View Only) by using another account.
> Check the table functionalities work as they should (starred, locked, duplicate, delete)
>Check the sign out button works 
>Check if you are only routed to the questionnaire when clicking on the questionnaire title and not the whole row
>Check the row border is orange when hovering over it 
> Check the new table component displays in storybook by running yarn storybook on eq-author
>For the pagination component: 
    (to test the different cases for multiple pages instead of duplicating questionnaires repeatedly you can change the PAGE_SIZE in components/QuestionnairesView/reducer.js)
- Check the text Page x of x displays the correct current page out of total pages
- When there are 7 pages check if the pagination displays buttons 1 to 7 (no ellipses)
- When there are more then 7 pages and your current page is less than 5 there should be an ellipses on the right hand side e.g 1 2 3 4 5 6 7 ... 20 (20 is the last page)
- When there are more then 7 pages and your current page is greater or equal to 5 there should be an ellipses on the left and right hand side e.g. 1 ... 3 4 5 6 7 ... 20
- When there are more then 7 pages and your current page is either of the last 4 pages the ellipses should display on the left hand side e.g. 1 ... 14 15 16 17 18 19 20
- Check your current page is focused on the correct pagination button 
- Check there is a light grey background when hovering over the pagination buttons
- Check the previous button is disabled when on the first page 
- Check the next button is disabled when on the last page.



### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
